### PR TITLE
movement modifier system

### LINKED ISSUE #592: You move at mach 5 after rejuvenating from starvation death.

### PR DESCRIPTION
**Pull request by 4dplanner**
fixes #592 

Adds an IMoveSpeedModifier interface that can alter walk/sprint speed for the object it's put on, if PlayerInputMoverComponent.MarkMovementSpeedModifiersDirty() is called.

### CODE REVIEW

File: Content.Server/GameObjects/Components/Movement/AiControllerComponent.cs [hunk 1]
```diff
@@ -60,13 +60,13 @@ public override void ExposeData(ObjectSerializer serializer)
...
         [ViewVariables(VVAccess.ReadWrite)]
-        public float WalkMoveSpeed { get; set; } = 4.0f;
+        public float CurrentWalkSpeed { get; set; } = 4.0f;
 
         /// <summary>
         ///     Movement speed (m/s) that the entity sprints.
         /// </summary>
         [ViewVariables(VVAccess.ReadWrite)]
-        public float SprintMoveSpeed { get; set; } = 10.0f;
+        public float CurrentSprintSpeed { get; set; } = 10.0f;
```
Comment (4dplanner) [inline Content.Server/GameObjects/Components/Movement/AiControllerComponent.cs]: Note: this is different to the Player one, is this intentional?

Comment (PJB3005) [reply Content.Server/GameObjects/Components/Movement/AiControllerComponent.cs]: Sounds like a case of "woops I didn't realize it's a duplicate value"

**Comment by 4dplanner** [comment]
Thinking it's probably best if I extract the slowdown stuff from the input component, will give it a go soon

**Comment by 4dplanner** [comment]
Going to re-do this with a slowdown component instead

**Pull request closed without merging**